mne/decoding/csp.py

```diff
@@ -402,20 +401,10 @@ def plot_patterns(
         fig : instance of matplotlib.figure.Figure
            The figure.
         """
-        if units is None:
-            units = "AU"
-        if components is None:
-            components = np.arange(self.n_components)
-
-        # set sampling frequency to have 1 component per time point
-        info = cp.deepcopy(info)
-        with info._unlock():
-            info["sfreq"] = 1.0
-        # create an evoked
-        patterns = EvokedArray(self.patterns_.T, info, tmin=0)
-        # the call plot_topomap
-        fig = patterns.plot_topomap(
-            times=components,
+        fig = _plot_model(
```

Comment:
Better to use public than private if possible. So could we do instead:
```
spatial_filter = get_spatial_filter_from(self)
return spatial_filter.plot_patterns(...)
```
?

Or is there some reason the private `_plot_model` is better / needed here?

Also ideally we'd decorate `CSP.plot_patterns` as `@legacy(alt="get_spatial_filter_from_est(clf).plot_patterns()"), assuming it is a better way of doing things!